### PR TITLE
fix: More instances of Avatar and booker links fixed

### LINKED ISSUE #10092: RFC: [CAL-2158] Use BOOKER_URL instead of WEBAPP_URL for booking pages links
Use BOOKER_URL instead of WEBAPP_URL to link to booker pages. Earlier WEBAPP_URL and BOOKER_URL were always same but now due to an org being able to have it's own subdomain(eg. [acme.cal.com](http://acme.cal.com)) for booker, this is not the case.

BOOKER_URL can't be a constant because it is derived from the slug set for the organization. So, this responsibility can be taken care of by  a new `useBookerUrl` hook which can give the URL(after deriving from useOrgBrandingValues)

Also, avatars are also supposed to be loaded from BOOKER_URL.

There are many many occurences where the instances are spread across the app. Goal is to be easily able to manage booking URL at a single place.

[Source image](https://github.com/calcom/cal.com/assets/1780212/ef40b419-33bc-4a68-8f7f-5c99021bc0f9)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2158](https://linear.app/calcom/issue/CAL-2158/use-booker-url-instead-of-webapp-url-for-booking-pages-links)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

- Updates more links of Avatars and booking pages with organization context.
- Introduces `useBookerUrl` to easily access the bookerUrl spread across the app. Would be easy to make fixes in it in future too.
- Also updates Reschedule, Cancel, Manage booking link sent in email with org context.
- Fixes missing translation on the Empty org booking page.
Fixes #10092

## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)
  
## How should this be tested?

- Test out different places where avatar is shown or the pages that take to the booking page

## Mandatory Tasks

  - [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/lib/server/getBrand.ts
Comment (hariombalhara) [inline packages/lib/server/getBrand.ts:1]: Created to be able to reuse on server.

File: apps/web/components/booking/BookingListItem.tsx [hunk 1]
```diff
@@ -170,7 +173,7 @@ function BookingListItem(booking: BookingItemProps) {
           id: "reschedule",
           icon: Clock,
           label: t("reschedule_booking"),
-          href: `/reschedule/${booking.uid}${
+          href: `${bookerUrl}/reschedule/${booking.uid}${
```
Comment (hariombalhara) [inline apps/web/components/booking/BookingListItem.tsx:176]: image

File: apps/web/pages/team/[slug].tsx [hunk 2]
```diff
@@ -152,9 +152,9 @@ function TeamPage({ team, isUnpublished, markdownStrippedBio, isValidOrgDomain }
         <div className="overflow-hidden rounded-sm border dark:border-gray-900">
           <div className="text-muted dark:text-inverted p-8 text-center">
             <h2 className="font-cal dark:text-inverted text-emphasis600 mb-2 text-3xl">
-              {" " + t("no_teams_yet")}
+              {" " + t("org_no_teams_yet")}
```
Comment (hariombalhara) [inline apps/web/pages/team/[slug].tsx:155]: These translations don't exist.

File: apps/web/pages/team/[slug].tsx [hunk 3]
```diff
@@ -152,9 +152,9 @@ function TeamPage({ team, isUnpublished, markdownStrippedBio, isValidOrgDomain }
         <div className="overflow-hidden rounded-sm border dark:border-gray-900">
           <div className="text-muted dark:text-inverted p-8 text-center">
             <h2 className="font-cal dark:text-inverted text-emphasis600 mb-2 text-3xl">
-              {" " + t("no_teams_yet")}
```
Comment (hariombalhara) [inline apps/web/pages/team/[slug].tsx:155]: These translations didn't eixst.

File: packages/lib/CalEventParser.ts [hunk 4]
```diff
@@ -174,7 +174,7 @@ export const getRescheduleLink = (calEvent: CalendarEvent): string => {
   const Uid = getUid(calEvent);
   const seatUid = getSeatReferenceId(calEvent);
 
-  return `${WEBAPP_URL}/reschedule/${seatUid ? seatUid : Uid}`;
+  return `${calEvent.bookerUrl ?? WEBAPP_URL}/reschedule/${seatUid ? seatUid : Uid}`;
```
Comment (hariombalhara) [inline packages/lib/CalEventParser.ts:177]: For Org
image

For non org
image

File: packages/lib/server/getBookerUrl.ts
Comment (hariombalhara) [inline packages/lib/server/getBookerUrl.ts:1]: Non Hook version for use on server.

File: packages/features/ee/organizations/components/TeamInviteFromOrg.tsx [hunk 5]
```diff
@@ -65,20 +65,21 @@ function UserToInviteItem({
...
+        "flex cursor-pointer items-center rounded-md px-2 py-1",
         isSelected ? "bg-emphasis" : "hover:bg-subtle "
       )}>
       <div className="flex items-center space-x-2 rtl:space-x-reverse">
         <Avatar
           size="sm"
           alt="Users avatar"
           asChild
-          imageSrc={`/api/user/${member.user.username}`}
+          imageSrc={`${bookerUrl}/${member.user.username}/avatar.png`}
```
Comment (hariombalhara) [inline packages/features/ee/organizations/components/TeamInviteFromOrg.tsx:82]: image

File: packages/features/ee/organizations/components/TeamInviteFromOrg.tsx [hunk 6]
```diff
@@ -65,20 +65,21 @@ function UserToInviteItem({
...
   onChange: () => void;
 }) {
+  const bookerUrl = useBookerUrl();
   return (
     <div
       key={member.userId}
       onClick={() => onChange()} // We handle this on click on the div also - for a11y we handle it with label and checkbox below
       className={classNames(
-        "flex cursor-pointer items-center rounded-md py-1 px-2",
+        "flex cursor-pointer items-center rounded-md px-2 py-1",
```
Comment (hariombalhara) [inline packages/features/ee/organizations/components/TeamInviteFromOrg.tsx:74]: Automatically done by prettier.

File: packages/features/ee/teams/components/AddNewTeamMembers.tsx [hunk 7]
```diff
@@ -214,7 +215,7 @@ const PendingMemberItem = (props: { member: TeamMember; index: number; teamId: n
         <Avatar
           gravatarFallbackMd5="teamMember"
           size="mdLg"
-          imageSrc={WEBAPP_URL + "/" + member.username + "/avatar.png"}
+          imageSrc={bookerUrl + "/" + member.username + "/avatar.png"}
```
Comment (hariombalhara) [inline packages/features/ee/teams/components/AddNewTeamMembers.tsx:218]: image

File: packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts [hunk 8]
```diff
@@ -142,7 +144,6 @@ export const getByViewerHandler = async ({ ctx }: GetByViewerOptions) => {
       teamId: group.teamId,
       ...group.profile,
       ...group.metadata,
-      image: `${CAL_URL}/${group.profile.slug}/avatar.png`,
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/webhook/getByViewer.handler.ts:145]: It's not needed. Bing already set above

**Review by leog — APPROVED** [review]
LGTM, well done refactoring the getBrand and configuring `staleTime`, had the latter in my TODO list

**Pull request merged**